worker/src/handles/UdpSocketHandle.cpp

```diff
@@ -322,6 +348,13 @@ bool UdpSocketHandle::SetLocalAddress()
 	Utils::IP::GetAddressInfo(
 	  reinterpret_cast<const struct sockaddr*>(&this->localAddr), family, this->localIp, this->localPort);
 
+	err = uv_fileno(reinterpret_cast<uv_handle_t*>(this->uvHandle), std::addressof(this->fd));
+
+	if (err != 0)
+	{
+		MS_ERROR("uv_fileno() failed: %s", uv_strerror(err));
```

Comment:
Why this? When can this fail? And what would happen if this fail? How is this related to liburing? Also, in p TcpSocketHandle.cpp` here you are also calling `return false;` but here that0s missing.

Reply:
> Why this? When can this fail? 

The method I'm calling to retrieve the file descriptor can fail, so we must handle. I should not fail but it can.

> And what would happen if this fail?

We should abort, I've added a `return false` right now.

> How is this related to liburing? 

libuv does know the file descriptor of it's handles, but in liburing we need to tell it which file descriptor we want to send data on.

> TcpSocketHandle.cpphere you are also callingreturn false;` but here that0s missing.

Fixed.

Reply:
> > And what would happen if this fail?
> 
> We should abort, I've added a `return false` right now.

But what are the implications of returning false? Or should we replace with MS_ABORT()?


> > How is this related to liburing?
> 
> libuv does know the file descriptor of it's handles, but in liburing we need to tell it which file descriptor we want to send data on.

I don't get it. I mean, I do, but why is this now needed? Literally in `UdpSocketHandle` the PR does this:

```c++
	Utils::IP::GetAddressInfo(
	  reinterpret_cast<const struct sockaddr*>(&this->localAddr), family, this->localIp, this->localPort);


	// This is new in this PR:
	err = uv_fileno(reinterpret_cast<uv_handle_t*>(this->uvHandle), std::addressof(this->fd));

	if (err != 0)
	{
		MS_ERROR("uv_fileno() failed: %s", uv_strerror(err));
```

Why isn't that protected by `#ifdef MS_LIBURING_SUPPORTED` macro?

Reply:
And why is that code in `SetLocalAddress()` method? That method is only supposed to obtain local IP and port in string/integer format, nothing else.

Reply:
> But what are the implications of returning false? Or should we replace with MS_ABORT()?

If `this.SetLocalAddress()` returns false, then the calling code will already call MS_ABORT.

>  but why is this now needed? 

We need the file descriptor of our local sockets in order to tell to liburing to send data over those file descriptors.

> Why isn't that protected by #ifdef MS_LIBURING_SUPPORTED macro?

I can put the macro sure, and also to the definition of `this->fd`. Do you want it?

> And why is that code in SetLocalAddress() method? That method is only supposed to obtain local IP and port in string/integer format, nothing else.

Let me put out of there.